sisyphus/simple_linux_utility_for_resource_management_engine.py

```diff
@@ -86,16 +86,17 @@ def system_call(self, command, send_to_stdin=None):
         """
         if self.gateway:
             escaped_command = [shlex.quote(s) for s in command]  # parameters need to be shell safe when sending via ssh
-            system_command = ["ssh", "-x", self.gateway] + [" ".join(["cd", os.getcwd(), "&&"] + escaped_command)]
+            system_command = ["ssh", "-x", self.gateway, "-o", "BatchMode=yes"] + [
+                " ".join(["cd", os.getcwd(), "&&"] + escaped_command)
+            ]
         else:
             # no gateway given, skip ssh local
             system_command = command
 
         logging.debug("shell_cmd: %s" % " ".join(system_command))
-        p = subprocess.Popen(system_command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         if send_to_stdin:
             send_to_stdin = send_to_stdin.encode()
-        out, err = p.communicate(input=send_to_stdin, timeout=30)
+        p = subprocess.run(system_command, input=send_to_stdin, capture_output=True, timeout=30)
```

Comment:
I wonder, don't you need to catch `TimeoutExpired`?

Reply:
Yes, you're right, I'll ignore any `TimeoutExpired`.

Reply:
So what does it mean? What happens now? The whole manager crashes?

Reply:
I don't think I had any `TimeoutExpired` exception, my error was within the timeout because of a public key mismatch, and it was actually outputted within the first second after the command was run:
```
[2024-06-13 07:24:26,842] INFO: Submit to queue: work/i6_core/returnn/extract_prior/ReturnnComputePriorJobV2.yYMFtGVTZgnn run [1]
[2024-06-13 07:24:26,953] ERROR: Error to submit job, return value: 255
```

Reply:
Actually, I realize now, in the code before your change, we also did not catch `TimeoutExpired` (in the `p.wait(timeout=30)`). I wonder why we did not catch it, and what happened when this exception was thrown there. Do you know? (As I understood you, you have run into this exact problem.)

Maybe we should not change this and also not catch `TimeoutExpired`? @critias?